tracing-subscriber/src/layer.rs

```diff
@@ -808,6 +808,92 @@ where
     }
 }
 
+impl<L, S> Layer<S> for Option<L>
+where
+    L: Layer<S>,
+    S: Subscriber,
+{
+    #[inline]
+    fn new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.new_span(attrs, id, ctx)
+        }
+    }
+
+    #[inline]
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        match self {
+            Some(ref inner) => inner.register_callsite(metadata),
+            None => Interest::always(),
+        }
+    }
+
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>, ctx: Context<'_, S>) -> bool {
+        match self {
+            Some(ref inner) => inner.enabled(metadata, ctx),
+            None => true,
```

Comment:
Should this be true, when `inner` is `None`?

Reply:
Yes, because returning `false` would disable the span or event. `enabled` for a nested stack of layers short-circuits on the first `false`.

This behavior will likely change in a future release when we introduce support for per-layer filters, but it's correct for now.

Reply:
Ahhh, That makes sense!